Basic wireframe for a new tab type

zzhaobraze opened the pull request:
# Pull Request/Issue Resolution

## Usage:

```
{% sdktabs %}
{% sdktab tab1 %}
Tab1 Content
{% endsdktab %}

{% sdktab Tab2 %}
Tab2 Content
{% endsdktab %}
{% endsdktabs %}
```

## Example:

![image](https://github.com/braze-inc/braze-docs/assets/38015013/f48b700e-df1a-422e-919b-e6959ac3fba5)

internetisaiah commented:
Notes:

1. The tab should still show even if there is only 1 `{% sdktabs %}` on the page. This is the default behavior on the existing `{% tabs %}` tag.
2. The new SDK tab is reading both `{% sdktabs %}` _and_ `{% tabs %}` as SDK tabs. Instead, regular tabs should be able to be fully nested inside SDK tabs. (See image below for existing behavior).
3. When an SDK tab is selected, the TOC should only show the headings for the currently selected SDK tab. Currently all headings are shown regardless of the SDK tab selected.

![image](https://github.com/user-attachments/assets/004c656b-9da0-4408-821b-53a206c96fe0)

cc @zzhaobraze

zzhaobraze commented:
@internetisaiah the previous commit should fix this issue. If you update the branch and resolve the conflict, this should work now.

internetisaiah commented:
@zzhaobraze , thanks for the update! i already merged it into my other branch on friday and have been testing it out. works great!

internetisaiah commented:
@zzhaobraze , similar to `{% tabs %}` and `{% subtabs %}`, could we get an `{% sdktabs %}` and `{% sdksubtabs %}`?

As far as design goes, it can look and function exactly like a regular tab.

However in terms of nesting, it should go:

- Highest level: `{% sdktabs %}`
- Next level: `{% sdksubtabs %}`
- Next level: `{% tabs %}`
- Lowest level: `{% subtabs %}`

internetisaiah commented:
ok @zzhaobraze , everything is looking amazing! 🙌🏽  thanks for getting it to the state it's in currently. i've only found one last "bug" that's a priority to fix. all the other bugs are low impact and can be tackled down the road though.

when setting up a redirect like this:

```js
validurls['/docs/developer_guide/platforms/web/sdk_integration/'] = '/docs/developer_guide/sdk_integration/?sdktab=web';
``` 

the redirect works, but it always selects the android tab by default (regardless of the URL ending). not sure if this helps troubleshoot, but this is the URL in the browser after the direct. not sure if its like an order of operations thing?

```
http://127.0.0.1:5006/docs/developer_guide/sdk_integration/?sdktab=swift?redirected=1
```

here's a screenshot of what i see in the browser:

![Screen Shot 2025-02-10 at 17 51 12 PM](https://github.com/user-attachments/assets/25fd36f5-6478-4455-8302-2dcd1d5f0c62)

internetisaiah commented:
@zzhaobraze , also, i updated the colors in [this commit](https://github.com/braze-inc/braze-docs/pull/7290/commits/639bf531d98a9df1c6a4f8bcb019fbdebe4acd43) to better match the standard tab colors (the previous colors didn't have as much contrast so i wanted to make them easier to read. let me know if i changed any of the wrong lines--i'm happy to fix it 👍🏽

zzhaobraze commented:
> @zzhaobraze , also, i updated the colors in [this commit](https://github.com/braze-inc/braze-docs/pull/7290/commits/639bf531d98a9df1c6a4f8bcb019fbdebe4acd43) to better match the standard tab colors (the previous colors didn't have as much contrast so i wanted to make them easier to read. let me know if i changed any of the wrong lines--i'm happy to fix it 👍🏽

i fix the redirect issue. was a bug due to the multi-redirect change. it should be an '&' instead of a '?'
i would make the contrast more obvious, let me know if you need help with that.

internetisaiah commented:
> If there is a question of contrast, can we loop Jucelle in to the conversation to make sure we're picking ADA compliant colors?

for now i've set the colors to match our existing tabs 1:1 (which i'm assuming already meet this standard), but i do plan on looping in a designer in the next few weeks to see if we can get some tweaks to the design (but that would be separate work that doesn't impact this PR from being merged).
